compiler/noirc_frontend/src/elaborator/types.rs

```diff
@@ -412,20 +410,57 @@ impl<'context> Elaborator<'context> {
                 let kind = self
                     .interner
                     .get_global_let_statement(id)
-                    .map(|let_statement| Kind::numeric(let_statement.r#type))
+                    .map(|let_statement| {
+                        if let_statement.r#type == Type::FieldElement {
+                            Kind::IntegerOrField
+                        } else {
+                            Kind::numeric(let_statement.r#type)
+                        }
+                    })
```

Comment:
What is this check for?

Reply:
It's a leftover method to ensure that field element's can be used polymorphically, but it's now unused so I've removed it.